float32 hip_x
float32 hip_y
float32 knee
